contracts/voting_escrow/src/utils.rs

```diff
@@ -158,3 +159,18 @@ pub(crate) fn validate_addresses(deps: Deps, addresses: &[String]) -> StdResult<
         .map(|addr| addr_validate_to_lower(deps.api, addr))
         .collect()
 }
+
+/// Calculate slashed and return amount based on a given parameters.
+/// The penalty is calculated as min(max_exit_penalty, time_left_until_unlock / MAX_LOCK_TIME).
+pub(crate) fn calc_early_withdraw_amount(
+    max_exit_penalty: Decimal,
+    periods_upon_unlock: u64,
+    xastro_amount: Uint128,
+) -> (Uint128, Uint128) {
+    let user_penalty = Decimal::from_ratio(periods_upon_unlock, get_periods_count(MAX_LOCK_TIME));
+    let exact_penalty = min(max_exit_penalty, user_penalty);
+    let slashed_amount = xastro_amount * exact_penalty;
```

Comment:
Let's use checked_mul to prevent getting this overflow error in a future audit.: https://github.com/astroport-fi/astroport-core/blob/7194e3b3e143ee5621102a2ff30a4a942dbfd773/packages/astroport/src/lib.rs#L38

Reply:
'exact_penalty' always lays in the [0, 1] interval. How would we get an overflow here?